packages/db/src/models/granule.ts

```diff
@@ -20,15 +21,20 @@ export default class GranulePgModel extends BasePgModel<PostgresGranule, Postgre
     executionCumulusId: number,
     granulesExecutionsPgModel = new GranulesExecutionsPgModel()
   ) {
+    if (!granule.created_at) {
+      throw new Error(`To upsert granule record must have 'created_at' set: ${JSON.stringify(granule)}`);
+    }
     if (granule.status === 'running') {
       return knexOrTrx(this.tableName)
         .insert(granule)
-        .onConflict(['granule_id', 'collection_cumulus_id'])
+        .onConflict(['granule_id', 'collection_cumulus_id']) // should unique_constraints be a method on the db model?
```

Comment:
I'm not opposed per se: but is that because that set of fields is used twice in this method?

Reply:
That's where that thought came from, yes. 

I'm going to err on the side of *not* doing that here, we can raise it again if a DRY concern comes up elsewhere. 

51341a025